onnxruntime/core/providers/coreml/model/model.mm

```diff
@@ -298,23 +354,23 @@ - (Status)predict:(const std::unordered_map<std::string, OnnxTensorData>&)inputs
         return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "output_features has no value for ", output_name);
       }
 
-      auto* data = [output_value multiArrayValue];
+      __block MLMultiArray* data = [output_value multiArrayValue];
 
-      const auto coreml_static_output_shape = [&]() {
+      const auto coreml_static_output_shape = [](MLMultiArray* data) {
```

Comment:
just curious, do `__block` variables work differently for lambda captures?

Reply:
yes, variables with `__block` storage cannot be captured by & from c++ lambdas (compile error). I believe the `__block` identifier simply isn't understood on the cpp side, as the behavior shouldn't be different outside objc blocks interactions (there may be objc compiler optimizations that prove me wrong here though).

I should say we don't really need __block here though as var isn't mutable, realized it while adding the "const" I overlooked earlier, so thank you! :)